data/registry_model/registry_oci_model.py

```diff
@@ -506,75 +451,15 @@ def change_repository_tag_expiration(self, tag, expiration_date):
         """
         return oci.tag.change_tag_expiration(tag._db_id, expiration_date)
 
-    def get_legacy_images_owned_by_tag(self, tag):
-        """
-        Returns all legacy images *solely owned and used* by the given tag.
-        """
-        tag_obj = oci.tag.get_tag_by_id(tag._db_id)
-        if tag_obj is None:
-            return None
-
-        tags = oci.tag.list_alive_tags(tag_obj.repository_id)
-        legacy_images = oci.tag.get_legacy_images_for_tags(tags)
-
-        tag_legacy_image = legacy_images.get(tag._db_id)
-        if tag_legacy_image is None:
-            return None
-
-        assert isinstance(tag_legacy_image, Image)
-
-        # Collect the IDs of all images that the tag uses.
-        tag_image_ids = set()
-        tag_image_ids.add(tag_legacy_image.id)
-        tag_image_ids.update(tag_legacy_image.ancestor_id_list())
-
-        # Remove any images shared by other tags.
-        for current in tags:
-            if current == tag_obj:
-                continue
-
-            current_image = legacy_images.get(current.id)
-            if current_image is None:
-                continue
-
-            tag_image_ids.discard(current_image.id)
-            tag_image_ids = tag_image_ids.difference(current_image.ancestor_id_list())
-            if not tag_image_ids:
-                return []
-
-        if not tag_image_ids:
-            return []
-
-        # Load the images we need to return.
-        images = database.Image.select().where(database.Image.id << list(tag_image_ids))
-        all_image_ids = set()
-        for image in images:
-            all_image_ids.add(image.id)
-            all_image_ids.update(image.ancestor_id_list())
-
-        # Build a map of all the images and their parents.
-        images_map = {}
-        all_images = database.Image.select().where(database.Image.id << list(all_image_ids))
-        for image in all_images:
-            images_map[image.id] = image
-
-        return [LegacyImage.for_image(image, images_map=images_map) for image in images]
-
     def get_security_status(self, manifest_or_legacy_image):
         """
         Returns the security status for the given manifest or legacy image or None if none.
         """
-        image = None
-
-        if isinstance(manifest_or_legacy_image, Manifest):
-            image = oci.shared.get_legacy_image_for_manifest(manifest_or_legacy_image._db_id)
-            if image is None:
-                return SecurityScanStatus.UNSUPPORTED
-        else:
-            try:
-                image = database.Image.get(id=manifest_or_legacy_image._db_id)
-            except database.Image.DoesNotExist:
-                return None
+        # TODO: change from using the Image row once we've moved all security info into MSS.
```

Comment:
Per our discussion, we'd actually want to remove this function completely and only access this information using the `secscan_model`.